### PR TITLE
(DOCSP-29409, DOCSP-29158, DOCSP-29157): Add Kotlin driver to Drivers…

### PR DESCRIPTION
**Pull request by cbullinger**
… site

# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-ecosystem/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-29409>
Staging - <https://docs-mongodbcom-staging.corp.mongodb.com/drivers/docsworker-xlarge/docsp-29409-kotlin-landing/kotlin/>

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [ ] Are all the links working?

### CODE REVIEW

File: source/kotlin-sync.txt
Comment (ccho-mongodb) [inline source/kotlin-sync.txt:79]: Question:
Could the placeholder be "\<connection string\>" instead, similar to how it is formatted on the Kotlin sync connection example?
Other drivers use that convention because the instructions mention retrieving the value from Atlas. This would also help clarify the text in lines 114-116:

> To test whether you can connect to your server, replace the connection
string in the :ref:`Connect to MongoDB Atlas <connect-atlas-kotlin-sync-driver>`
code example and run it.

**Review by ccho-mongodb — CHANGES_REQUESTED** [review]
Looks great, just a couple small issues/suggestions.

File: source/kotlin-sync.txt
Comment (cbullinger) [reply source/kotlin-sync.txt:47]: Good catch. I've updated to use a source constant, but it is a little awkward until the Java and Kotlin driver versions are in sync.

File: source/kotlin-sync.txt
Comment (cbullinger) [reply source/kotlin-sync.txt:18]: Updated verbiage

File: source/kotlin-sync.txt
Comment (cbullinger) [reply source/kotlin-sync.txt:15]: Updated

File: source/kotlin-sync.txt
Comment (cbullinger) [reply source/kotlin-sync.txt:16]: Updated

File: source/kotlin-sync.txt
Comment (ccho-mongodb) [reply source/kotlin-sync.txt:13]: I guess it depends on the interpretation of the style guide guidance. In any case, here's a recent example of an warning admonition in which we presented a product that is not quite ready to recommend for production deployments:

https://www.mongodb.com/docs/manual/core/queryable-encryption/

File: snooty.toml
Comment (ccho-mongodb) [inline snooty.toml:17]: Suggestion:

If the PR is meant to be merged and published when the Java v4.10 drivers are released, it would probably be fine to update the java-version source constant to "4.10" instead of creating a "-next" and referencing it in "-full". If it is meant to be merged prior, then feel free to ignore.

**Review by ccho-mongodb — APPROVED** [review]
LGTM, just added one suggestion if this PR is meant to be published at the same time as Java v4.10 drivers become available.

File: snooty.toml
Comment (cbullinger) [reply snooty.toml:17]: I thought this was going to be published before Java v4.10, but now it's looking like we're putting on hold until Kotlin GA / Java v4.10 release -- Ashni to confirm. If that's the case, I'll strip the '(Alpha)` label and update these source constants, then wait to merge until the GA release.

**Review by ccho-mongodb — APPROVED** [review]
LGTM

**Pull request merged**
